### DISCUSSION TITLE
How to specify a view mode when opening the history of a file from the command line

### DISCUSSION THREAD

**Discussion by alewis001** [jesseduffield/lazygit#4011] [Q&A] [answered]
if I go to the Commits panel, do ctrl+s and find a file, the view that appears is "Commits / Reflog" on the left and "Patch" on the right (showing the commit patch preview for the file I selected in the filter); I think this the "half" view mode but I might have that wrong.

If I do `lazygit -f <file>`, then instead of the "half" view mode like above I get the mode where only the commit history is shown. I can obviously cycle through the view modes with `+` but it would be nice if I could shortcut that in some way.

**Is it possible to specify the view mode when using `-f`?**

The context to me wanting to do this is that I use Lazygit inside Lazyvim. LV uses the `-f` flag to open up the file history view. I'd like to be able to say use "half" view mode by default.

Thinking more generally, I can see how using `lazygit -f <file` from the command line outside of lazyvim and have it open the view in "half" mode would also be nice (I didn't know about `-f` until today :) ). I now especially like `lazygit -f **<tab>` to open `fzf` to find the file I want to view the history of :)

If the functionality doesn't exist today then some ideas could be...

* A new command line arg
    * Would solve my problem in Lazyvim and on the command line but you'd have to remember to use the arg, more typing, etc. and we are trying to be "lazy" after all :)
* Config to specify the default view mode for the file history view
    * Could combine with the command line arg with the the arg taking precedence over the default in the config

Any other ideas?

I could give contributing a PR a go if there's a consensus on how it should be done.

**Comment by knowald** [comment]
+1 here. Same scenario how i arrived at this question since i'm using the [lazyvim](https://github.com/LazyVim/LazyVim) preset, but setting a default view mode would be very nice to have in general. Thanks for mentioning the `+` view mode cycle, because i somehow failed to spot that option and couldn't figure out why there's a viewing difference between opening a file with `-f` compared to opening `lazygit` normally and filtering for the file.

**Comment by alewis001** [comment]
Hi @jesseduffield, I hope you don't mind me pinging you here. 

I've started work on a change to introduce a `-sm normal|half|full` / `--screen-mode normal|half|full` command line argument. My changes are here: https://github.com/jesseduffield/lazygit/compare/master...alewis001:lazygit:cmd-line-view-mode. 

Would it be possible for you to take a quick look and provide any guidance on whether you'd be willing to accept a PR for this change, am I going in the right direction or would you prefer another, etc? 

Regarding testing, what would you want to see in a PR? I did have a look at doing integration tests but wasn't sure how to validate the resulting screen mode in lazygit matched what was provided on the command line. I'll keep looking through the existing tests to see if I can work it out in the meantime.

In general, it's a very early cut but wanted to get some early feedback so I'd really appreciate any that you have, good or bad :)

Thanks

**Comment by alewis001** [comment, marked as answer]
My PR for this change has been merged: https://github.com/jesseduffield/lazygit/pull/4103

I'm going to close this discussion now :)
